clang/lib/Driver/Driver.cpp

```diff
@@ -9506,6 +9521,24 @@ const char *Driver::GetNamedOutputPath(Compilation &C, const JobAction &JA,
   else
     BaseName = llvm::sys::path::filename(BasePath);
 
+  // When compiling with -fsyclbin, maintain a simple output file name for the
+  // resulting image.  A '.syclbin' extension is intended to be added during
+  // the linking step with the clang-linker-wrapper.
+  if (JA.getOffloadingDeviceKind() == Action::OFK_SYCL &&
+      C.getArgs().hasArgNoClaim(options::OPT_fsyclbin) &&
+      JA.getType() == types::TY_Image) {
+    const char *SYCLBinOutput;
+    if (IsCLMode()) {
+      // clang-cl uses BaseName for the executable name.
+      SYCLBinOutput =
```

Comment:
Can you add a test case for this and L9537?

Reply:
I have modified some of the behaviors here to be more in line with expectations, for the output to be a `.syclbin` based file.